### PR TITLE
pack: make in link checking generic

### PR DESCRIPTION
**Pull request by mmcgee-jump**

### CODE REVIEW

File: src/app/fdctl/run/tiles/fd_pack.c [hunk 1]
```diff
@@ -677,6 +675,24 @@ unprivileged_init( fd_topo_t *      topo,
...
+    else if( FD_LIKELY( !strcmp( link->name, "bank_poh"    ) ) ) ctx->in_kind[ i ] = IN_KIND_BANK;
+    else if( FD_LIKELY( !strcmp( link->name, "bundle_pack" ) ) ) ctx->in_kind[ i ] = IN_KIND_BUNDLE;
+    else FD_LOG_ERR(( "pack tile has unexpected input link %lu %s", i, link->name ));
+  }
+
+  ulong out_cnt = fd_topo_link_consumer_cnt( topo, &topo->links[ tile->out_link_id_primary ] );
+
+  if( FD_UNLIKELY( !out_cnt ) ) FD_LOG_ERR(( "pack tile connects to no banking tiles" ));
+  if( FD_UNLIKELY( out_cnt>FD_PACK_PACK_MAX_OUT ) ) FD_LOG_ERR(( "pack tile connects to too many banking tiles" ));
+  if( FD_UNLIKELY( out_cnt!=tile->pack.bank_tile_count+1UL ) ) FD_LOG_ERR(( "pack tile connects to %lu banking tiles, but tile->pack.bank_tile_count is %lu", out_cnt, tile->pack.bank_tile_count ));
```
Comment (ptaffet-jump) [inline src/app/fdctl/run/tiles/fd_pack.c:694]: I know it's not part of this PR (probably my mistake), but could you align these and then fix the last one?
It should be `out_cnt-1UL` in the LOG_ERR line

File: src/app/fdctl/run/tiles/fd_pack.c [hunk 2]
```diff
@@ -677,6 +675,24 @@ unprivileged_init( fd_topo_t *      topo,
...
+
+  for( ulong i=0UL; i<tile->in_cnt; i++ ) {
+    fd_topo_link_t const * link = &topo->links[ tile->in_link_id[ i ] ];
+
+    if( FD_LIKELY(      !strcmp( link->name, "dedup_pack"  ) ) ) ctx->in_kind[ i ] = IN_KIND_DEDUP;
+    else if( FD_LIKELY( !strcmp( link->name, "poh_pack"    ) ) ) ctx->in_kind[ i ] = IN_KIND_POH;
+    else if( FD_LIKELY( !strcmp( link->name, "bank_poh"    ) ) ) ctx->in_kind[ i ] = IN_KIND_BANK;
+    else if( FD_LIKELY( !strcmp( link->name, "bundle_pack" ) ) ) ctx->in_kind[ i ] = IN_KIND_BUNDLE;
+    else FD_LOG_ERR(( "pack tile has unexpected input link %lu %s", i, link->name ));
+  }
```
Comment (ptaffet-jump) [inline src/app/fdctl/run/tiles/fd_pack.c:688]: I think this does eliminate some sanity checks, e.g. that there are  dedup->pack and poh->pack links

File: src/app/fdctl/run/tiles/fd_pack.c [hunk 3]
```diff
@@ -606,13 +616,22 @@ after_frag( void *             _ctx,
...
-  if( FD_UNLIKELY( in_idx==POH_IN_IDX ) ) {
+  switch( ctx->in_kind[ in_idx ] ) {
+  case IN_KIND_POH: {
     ctx->slot_end_ns = ctx->_slot_end_ns;
     fd_pack_set_block_limits( ctx->pack, ctx->slot_max_microblocks, ctx->slot_max_data );
-  } else if( FD_UNLIKELY( in_idx>=BANK_BASE_IN_IDX ) ) {
+    break;
+  }
+  case IN_KIND_BUNDLE: {
+    FD_LOG_WARNING(( "Pack tile received a bundle... dropping..." ));
```
Comment (ptaffet-jump) [inline src/app/fdctl/run/tiles/fd_pack.c:626]: This should be unreachable, but I guess it's fine to leave it as a placeholder

**Review by ptaffet-jump — COMMENTED** [review]
.

File: src/app/fdctl/run/tiles/fd_pack.c (hunk 2 above)
Comment (mmcgee-jump) [reply src/app/fdctl/run/tiles/fd_pack.c:688]: That's OK, the tile shouldn't really assume a topology in that way

**Review by ptaffet-jump — APPROVED** [review]

**Pull request merged**